src/v/archival/ntp_archiver_service.cc

```diff
@@ -362,9 +362,9 @@ ss::future<> ntp_archiver::upload_until_abort() {
         vlog(_rtclog.debug, "upload loop starting in term {}", _start_term);
         auto sync_timeout = config::shard_local_cfg()
                               .cloud_storage_metadata_sync_timeout_ms.value();
-        bool is_synced = co_await _parent.archival_meta_stm()->sync(
+        auto is_synced = co_await _parent.archival_meta_stm()->sync(
           sync_timeout);
-        if (!is_synced) {
+        if (!is_synced.has_value()) {
```

Comment:
`is_synced` isn't really a good name now, right? it's also odd to have a optional engaged check, but ignore the stored value. per Andrew's comment, is that changing in a subsequent PR?